### PR TITLE
1.7 compat

### PR DESCRIPTION
**Pull request by mmiller-max**
Testing full CI. Needs a bit of work (one failing test I think) and tidying

### CODE REVIEW

**Comment by mmiller-max** [comment]
Well the release of 1.7 forced my hand a bit with this one! Looks good to go though.

Its not super pretty but it replicates the same behaviour as what we are doing on 1.6. Also bonus is that it passes on nightly.

Lets keep 1.0 compat for now, although it may be worth removing it to overhaul the code somewhat.

File: src/compat_check.jl [hunk 1]
```diff
@@ -151,10 +154,11 @@ function get_dep_entries end
             if haskey(getdeps(active_env.manifest), testreport_proj.deps[dep])
                 push!(deps_to_check, getdeps(active_env.manifest)[testreport_proj.deps[dep]])
             else
+                version_number = isa(testreport_proj.compat[dep], String) ? VersionNumber(testreport_proj.compat[dep]) : VersionNumber(testreport_proj.compat[dep].str)
```
Comment (oxinabox) [inline src/compat_check.jl:157]: Can we do something like
```suggestion
                version_number = string(testreport_proj.compat[dep])
```

**Review by oxinabox — APPROVED** [review]

File: src/compat_check.jl (hunk 1 above)
Comment (mmiller-max) [reply src/compat_check.jl:157]: In 1.7 `VersionNumber(testreport_proj.compat[dep]` is a `Pkg.Types.Compat` which is a new type that didn't exist, so `string(testreport_proj.compat[dep])` just produces something like `"Compat(VersionSpec(\"1.0\"), \"1.0\")"`. However I can make this line clearer

Comment (mmiller-max) [reply src/compat_check.jl:157]: Oh I see you mean not use `VersionNumber`. Yes quite possibly. I've been thinking about this and it feels like there must be a better solution/I'm not sure this is working exactly as intended. For this PR I'll keep using `VersionNumber` as it keeps the behaviour consistent, but I think a better way may be to to use `VersionSpec`.

Comment (mmiller-max) [reply src/compat_check.jl:157]: #79

**Pull request merged**